lib/tessel/wifi.js

```diff
@@ -96,10 +96,30 @@ Tessel.prototype.connectToNetwork = function(opts) {
             .then(function(remoteProcess) {
               // Once the wifi restart process closes
               return self.receive(remoteProcess).then(function() {
-                logs.info('Credentials set!');
-                // End the connection
-                return self.connection.end()
-                  .then(resolve);
+                logs.info('Credentials set. Checking connection...');
+                self.connection.exec(commands.ubusListen())
+                .then(function(remoteProcess) {
+                  remoteProcess.stdout.on("data", function (data) {
+                    if(data.toString().indexOf("ifup") > -1) {
+                      logs.info("Successfully connected!");
+                      // End the connection
+                      return self.connection.end()
+                        .then(resolve);
+                    }
+                  });
+                  remoteProcess.stderr.on("data", function (data) {
+                    logs.info("Error connecting:", data.toString());
+                    // End the connection
+                    return self.connection.end()
+                      .then(resolve);
+                  });
+                  setTimeout(function () {
+                    logs.info("Timed out waiting to verify connection. Run `t2 wifi` to manually verify connection. If not connected, ensure you have entered the correct network credentials.");
+                    // End the connection
+                    return self.connection.end()
+                      .then(resolve);
+                  }, 10000);
```

Comment:
This will need to be cancelled otherwise it will always be called no matter what. I suggest moving it to before the `self.receive(...)` stuff, then do: 

```js
var timeout = setTimeout(...);

self.receive(remoteProcess).then(function() {
  // do the check for ifup etc here, then just...

  clearTimeout(timeout);
  return resolve();
}).catch(function(error) {
  logs.info("Error connecting:", error.toString()); // 
  clearTimeout(timeout);

  // this is sufficient, all connections will be closed by the process that called this operation
  return resolve();
});
```